src/main/groovy/util/StockMovementContext.groovy

```diff
@@ -0,0 +1,58 @@
+package util
+
+import org.pih.warehouse.api.StockMovementDirection
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.requisition.RequisitionSourceType
+import org.pih.warehouse.shipping.Shipment
+
+class StockMovementContext {
+    Order order
+
+    Shipment shipment
+
+    Requisition requisition
+
+    Location origin
+
+    Location destination
+
+    StockMovementDirection getStockMovementDirection(Location currentLocation) {
+        if (currentLocation == origin) {
+            return StockMovementDirection.OUTBOUND
+        }
+        if (currentLocation == destination || origin?.isSupplier()) {
+            return StockMovementDirection.INBOUND
+        }
+        return null
+    }
+
+    boolean isInbound() {
+        Location currentLocation = AuthService.currentLocation
+        return getStockMovementDirection(currentLocation) == StockMovementDirection.INBOUND
+    }
+
+    boolean isOutbound() {
+        Location currentLocation = AuthService.currentLocation
+        return getStockMovementDirection(currentLocation) == StockMovementDirection.OUTBOUND
+    }
+
+    boolean isCurrentLocationDownstreamConsumer() {
+        Location currentLocation = AuthService.currentLocation
+        return currentLocation.isDownstreamConsumer()
+    }
+
+    boolean isReturn() {
+        return order?.orderType?.isReturnOrder()
+    }
+
+    Boolean isFromPurchaseOrder() {
+        return shipment?.isFromPurchaseOrder
+    }
+
+    Boolean isElectronicType() {
+        return requisition?.sourceType == RequisitionSourceType.ELECTRONIC
```

Comment:
I think it would be better to add this method to `requisition` in the same way it is implemented for `OutboundStockMovement`/`OutboundStockMovementListItem` (initially I wanted to say, that you have this method implemented in the `Requisition` domain, but apparently I can't find it there.